packages/p2p/src/components/my-ads/my-ads-delete-modal.jsx

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Button, Modal, Text } from '@deriv/components';
+import { isDesktop, useIsMounted } from '@deriv/shared';
+import { observer } from 'mobx-react-lite';
+import { Localize } from 'Components/i18next';
+import { requestWS } from 'Utils/websocket';
+import FormError from 'Components/form/error.jsx';
+import { useStores } from 'Stores';
+import 'Components/my-ads/my-ads-delete-modal.scss';
+
+const MyAdsDeleteModal = () => {
+    const { my_ads_store } = useStores();
+    const isMounted = useIsMounted();
+    const [error_message, setErrorMessage] = React.useState('');
+
+    const onClickCancel = () => {
+        my_ads_store.setSelectedAdId('');
+        my_ads_store.setIsDeleteModalOpen(false);
+    };
+
+    const onClickConfirm = () => {
+        requestWS({ p2p_advert_update: 1, id: my_ads_store.selected_ad_id, delete: 1 }).then(response => {
+            if (isMounted()) {
+                if (response.error) {
+                    setErrorMessage(response.error.message);
+                } else {
+                    // remove the deleted ad from the list of items
+                    const updated_items = my_ads_store.adverts.filter(ad => ad.id !== response.p2p_advert_update.id);
+                    my_ads_store.setAdverts(updated_items);
+                    my_ads_store.setIsDeleteModalOpen(false);
+                }
+            }
+        });
```

Comment:
Is it possible to move this logic into `my_ads_store`?

Reply:
We can but this is only specific to this component and won't be used elsewhere so I left it here to make our store manageable. :)